pythonlib/src/mill/pythonlib/PublishModule.scala

```diff
@@ -0,0 +1,276 @@
+package mill.pythonlib
+
+import mill.api.Result
+import mill._
+
+/**
+ * A python module which also defines how to build and publish source distributions and wheels.
+ */
+trait PublishModule extends PythonModule {
+
+  // export these to make using them in subclasses possible without an import
+  type License = scalalib.publish.License
+  val License = scalalib.publish.License
+  type PublishMeta = PublishModule.PublishMeta
+  val PublishMeta = PublishModule.PublishMeta
+  type Developer = PublishModule.Developer
+  val Developer = PublishModule.Developer
```

Comment:
Should we export these from `mill.pythonlib`'s `package` object instead? We don't really do this kind of trait re-exporting elsewhere in Mill, so having people import from `mill.pythonlib`. https://github.com/com-lihaoyi/mill/issues/3570 is a problem, but I think better to just suffer that while we figure out a wholistic solution rather than doing ad-hoc mitigations for now

Reply:
sounds good, I'll move it to a package oject